content/technotes/2020-02-26-tic-ghactions.md

```diff
@@ -0,0 +1,188 @@
+- No restrictions on the available build platforms
+- No issues related to webhooks of third-party providers
+- CI settings configurable via the repository settings
+
+However, R was again not among the languages with native support/examples, so it was not easy to get started.
+After a first shot from [Max Held](https://github.com/maxheld83) via the [ghactions](https://github.com/maxheld83/ghactions) package, [Jim Hester](https://github.com/jimhester) sat down again and created the [setup-r](https://github.com/r-lib/actions/tree/master/setup-r) action which installs R on all available platforms with the option to specify different R versions.
+
+While the setup-r GitHub action is already usable, it's still in its early days and will mature over time.
+In tic we use [r-lib/setup-r](https://github.com/r-lib/actions/tree/master/setup-r) as the base combined with [r-lib/setup-pandoc](https://github.com/r-lib/actions/tree/master/setup-pandoc) and [r-lib/setup-tinytex](https://github.com/r-lib/actions/tree/master/setup-tinytex).
+We apply some opinionated changes on top, which we think enhance the CI setup substantially.
+In the following sections we explain these changes to be as transparent as possible.
+
+#### Caching
+
+When testing a package, by default [r-lib/actions](https://github.com/r-lib/actions) caches only **direct dependencies** and only on a **successful run**.
+This is done via action [actions/cache](https://github.com/actions/cache) by hashing a list of all package dependencies:
+
+
+```yml
+- name: Query dependencies
+  run: |
+    install.packages('remotes')
+    saveRDS(remotes::dev_package_deps(dependencies = TRUE), "depends.Rds", version = 2)
+  shell: Rscript {0}
+
+- name: Cache R packages
+  if: runner.os != 'Windows'
+  uses: actions/cache@v1
+  with:
+    path: ${{ env.R_LIBS_USER }}
+    key: ${{ runner.os }}-r-${{ matrix.config.r }}-${{ hashFiles('depends.Rds') }}
+    restore-keys: ${{ runner.os }}-r-${{ matrix.config.r }}-
+```
+
+tic instead caches **the complete R library** of the run **on every run** by using a forked version of [actions/cache](https://github.com/actions/cache): [pat-s/always-upload-cache](https://github.com/pat-s/always-upload-cache).
+
+These differences have the advantage that packages which are needed for side-effects (like building a pkgdown site or running codecov) are cached too.
+Because caches are immutable, tic (needs to) rebuilds the cache **daily**.
+If this didn't happen, every runner would otherwise always use the first cache forever.
+When packages are updated on CRAN, tic would download the cache first but then update those before proceeding to the "script" stage.
+After some time all packages from the cache would be outdated, making the cache useless.
+Hence, tic rebuilds the cache daily.
+Users can change this time frequency to their liking by altering the bash code that is used to generate the timestamps (within the `run` line of the following code block):
+
+```yml
```

Comment:
in the fix I made to master aliases won't work so this should be "yaml" not "yml". Thanks!